Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py

```diff
@@ -39,6 +39,12 @@ def update_incident_with_required_keys(incidents: List, required_keys: List):
         :param required_keys: keys need to be updated
 
     """
+    for incident in incidents:
+        res = demisto.executeCommand("GetIncidentsByQuery", {"query": f"id:({incident['id']})"})
+        # If an incident is deleted, remove it from the list to avoid a key error
+        if res[0].get('Contents') == '[]':
+            incidents.remove(incident)
```

Comment:
Modifying a list when iterating on it is a bad practice, consider

```python
incidents = [incident for incident in incidents if demisto.executeCommand("GetIncidentsByQuery", {"query": f"id:({incident['id']})"})[0].get("Contents") != "[]"]
```
or the non-comprehension equivalent, which would be easier to debug. 

(sorry for not mentioning it before)

Reply:
You are absolutely right, this is an important point.